scripts/check_referenced_files.py

```diff
@@ -0,0 +1,109 @@
+import os
+import argparse
+import yaml
+import logging
+
+#
+# Script that parses all yaml files in actions and rules and checks
+# that all* referenced files exist. This script assumes that
+# action/rule/workflow name and file name are the same, e.g.
+# Workflow name: ngi_uu_workflow
+# File name: ngi_uu_workflow.yaml
+#
+# Keys searched for are specifically:
+# In actions:
+# - entry_point
+# - default (parameters -> workflow -> default)
+#
+# In workflows:
+# - action (workflows -> main -> tasks -> <task_name> -> action)
+#
+# In rules:
+# - ref (action -> ref)
+#
+# * Not really all:
+# TODO: Referenced actions should probably be compared with action names rather
+# than file name. Because of the current implementation sensors are not included
+# in check.
+
+def find(d, tag):
+    if tag in d:
+        logger.debug("Found tag: {}, with value: {}".format(tag,d[tag]))
+        yield d[tag]
+    for k, v in d.items():
+        if isinstance(v, dict):
+            for i in find(v, tag):
+                yield i
+
+def check_action(pack_location, file, action):
+    if os.path.isfile(os.path.join(pack_location,"actions","{}.yaml".format(action))):
+        logger.debug("In file {}: {} exists".format(file, action))
+    else:
+        logger.error("In file {}: {} does not exist!".format(file, action))
+
+def check_file(pack_location, parsed_file, referenced_file):
+    if os.path.isfile(os.path.join(pack_location,"actions",referenced_file)):
+        logger.debug("In file {}: {} exists".format(parsed_file, referenced_file))
+    else:
+        logger.error("In file {}: {} does not exist!".format(parsed_file, referenced_file))
+
+def check_files_in_folder(folder_path, tags):
+    for file in os.listdir(os.path.join(pack_location,folder_path)):
+        logger.debug("Found file: {} in {}".format(file, folder_path))
+        if file.endswith(".yaml"):
+            logger.debug("Found file, {}, is a yaml file".format(file))
+            full_path_file = os.path.join(pack_location,folder_path,file)
+            yaml_dict = yaml.load(open(full_path_file))
```

Comment:
I've recently realized that we should use `safe_load`, rather than `load` in these cases for security reasons.